### PR TITLE
FF131 Relnote - Javascript iterator helpers

### PR DESCRIPTION
**Pull request by hamishwillee**
FF131 ships the TC39 [iterator helper proposal](https://github.com/tc39/proposal-iterator-helpers) in  https://bugzilla.mozilla.org/show_bug.cgi?id=1896390

This adds a release note.

Related docs work can be tracked in #35698

### CODE REVIEW

File: files/en-us/mozilla/firefox/releases/131/index.md [hunk 1]
```diff
@@ -22,6 +22,8 @@ This article provides information about the changes in Firefox 131 that affect d
 
 ### JavaScript
 
+- Support for synchronous iterator helpers, including: {{jsxref("Iterator.prototype.drop()")}}, {{jsxref("Iterator.prototype.every()")}}, {{jsxref("Iterator.prototype.filter()")}}, {{jsxref("Iterator.prototype.find()")}}, {{jsxref("Iterator.prototype.flatMap()")}}, {{jsxref("Iterator.prototype.forEach()")}}, {{jsxref("Iterator.prototype.map()")}}, {{jsxref("Iterator.prototype.reduce()")}}, {{jsxref("Iterator.prototype.some()")}} , {{jsxref("Iterator.prototype.take()")}}. These allow `Array`-like operations on large (and possibly infinite) data structures, without having to create intermediate `Array` objects, and in cases where creating such an object would not be possible. For more information see [Iterator helpers](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator#iterator_helpers) in the `Iterator` interface. ([Firefox bug 1896390](https://bugzil.la/1896390)).
```
Comment (bsmth) [inline files/en-us/mozilla/firefox/releases/131/index.md:25]: Minor one:

```suggestion
- Support for synchronous iterator helpers, including: {{jsxref("Iterator.prototype.drop()")}}, {{jsxref("Iterator.prototype.every()")}}, {{jsxref("Iterator.prototype.filter()")}}, {{jsxref("Iterator.prototype.find()")}}, {{jsxref("Iterator.prototype.flatMap()")}}, {{jsxref("Iterator.prototype.forEach()")}}, {{jsxref("Iterator.prototype.map()")}}, {{jsxref("Iterator.prototype.reduce()")}}, {{jsxref("Iterator.prototype.some()")}}, and {{jsxref("Iterator.prototype.take()")}}. These allow `Array`-like operations on large (and possibly infinite) data structures, without having to create intermediate `Array` objects, and in cases where creating such an object would not be possible. For more information see [Iterator helpers](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator#iterator_helpers) in the `Iterator` interface. ([Firefox bug 1896390](https://bugzil.la/1896390)).
```

And small q: what's the object in "creating such an object"?

Comment (hamishwillee) [reply files/en-us/mozilla/firefox/releases/131/index.md:25]: > And small q: what's the object in "creating such an object"?

Good question

```suggestion
- Support for synchronous iterator helpers, including: {{jsxref("Iterator.prototype.drop()")}}, {{jsxref("Iterator.prototype.every()")}}, {{jsxref("Iterator.prototype.filter()")}}, {{jsxref("Iterator.prototype.find()")}}, {{jsxref("Iterator.prototype.flatMap()")}}, {{jsxref("Iterator.prototype.forEach()")}}, {{jsxref("Iterator.prototype.map()")}}, {{jsxref("Iterator.prototype.reduce()")}}, {{jsxref("Iterator.prototype.some()")}}, and {{jsxref("Iterator.prototype.take()")}}. These allow `Array`-like operations on iterators without having to create intermediate `Array` objects, and support the case where iterators span very large data sets, and creating an `Array` would not be possible. For more information see [Iterator helpers](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator#iterator_helpers) in the `Iterator` interface. ([Firefox bug 1896390](https://bugzil.la/1896390)).
```

**Comment by hamishwillee** [comment]
Thanks for the review!

File: files/en-us/mozilla/firefox/releases/131/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/mozilla/firefox/releases/131/index.md]: super, thanks 👍🏻

**Review by bsmth — APPROVED** [review]
Thank you! Merging shortly 👍🏻

**Pull request merged**